File: furdega/Services/HomePage/Sections/CompanyBenefits/Dtos/Input/CompanyBenefitsSectionRequest.cs
using System.Linq;
using Furdega.Services.FileManagers;

namespace Furdega.Services.HomePage.Sections.CompanyBenefits.Dtos.Input
{
    public class CompanyBenefitsSectionRequest: HomeSectionModelBase, IRequestWithImage
    {
        public CompanyBenefitRequest CompanyBenefit1 { get; set; }
        public CompanyBenefitRequest CompanyBenefit2 { get; set; }
        public CompanyBenefitRequest CompanyBenefit3 { get; set; }
        public CompanyBenefitRequest CompanyBenefit4 { get; set; }

        public bool IsFilesExtensionCorrect()
        {
            var images = AllImages.Where(s => !string.IsNullOrEmpty(s?.Base64ImageString));

            return !images.Any() || images.Any() && images.All(s => s.IsFileExtensionCorrect());
        }

        public bool IsAllBase64ImagesExist() => AllImages.All(s => !string.IsNullOrEmpty(s?.Base64ImageString));

        private Image[] AllImages => new[]
        {
            CompanyBenefit1.Image,
            CompanyBenefit2.Image,
            CompanyBenefit3.Image,
            CompanyBenefit4.Image
        };
    }
}

File: furdega/Frontend/src/types/company-benefits-section/company-benefits-section-request.ts
import { HomeSectionBase } from "../home-section-base";
import { CompanyBenefitRequest } from './company-benefit-request'

export type CompanyBenefitsSectionRequest = HomeSectionBase & {
    companyBenefit1: CompanyBenefitRequest
    companyBenefit2: CompanyBenefitRequest
    companyBenefit3: CompanyBenefitRequest
    companyBenefit4: CompanyBenefitRequest
}